### PR TITLE
fix(ivy): destroy injector when module is destroyed

### PR DESCRIPTION
**Pull request by crisbeto**
Destroys the module's injector when an `NgModule` is destroyed which in turn calls the `ngOnDestroy` methods on the instantiated providers.

This PR resolves FW-739.

### CODE REVIEW

File: packages/core/src/di/r3_injector.ts [hunk 1]
```diff
@@ -101,7 +101,8 @@ export class R3Injector {
   /**
    * Flag indicating that this injector was previously destroyed.
    */
-  private destroyed = false;
+  get destroyed(): boolean { return this._destroyed; }
```
Comment (mhevery) [inline packages/core/src/di/r3_injector.ts]: Why turning this into getter?

**Review by mhevery — APPROVED** [review]

File: packages/core/src/di/r3_injector.ts (hunk 1 above)
Comment (crisbeto) [reply packages/core/src/di/r3_injector.ts]: I want to prevent people from being able to set it from the outside. It could also be turned into a `isDestroyed` method which'll generate slightly less ES5 code.

File: packages/elements/test/slots_spec.ts [hunk 2]
```diff
@@ -39,7 +39,10 @@ if (browserDetection.supportsCustomElements && browserDetection.supportsShadowDo
 
     afterAll(() => {
       destroyPlatform();
-      testContainer.remove();
+
+      if (testContainer.parentNode) {
+        testContainer.parentNode.removeChild(testContainer);
+      }
```
Comment (petebacondarwin) [inline packages/elements/test/slots_spec.ts]: Why this change? Does calling `testContainer.remove()` not remove it from its parent? That would seem like an obvious thing for such a method to do...

File: packages/core/src/render3/node_manipulation.ts [hunk 3]
```diff
@@ -456,6 +457,7 @@ function removeListeners(lView: LView): void {
       }
     }
     lView[CLEANUP] = null;
+    tView.cleanup = null;
```
Comment (kara) [inline packages/core/src/render3/node_manipulation.ts]: Hmm this doesn't look quite right since `TViews` are shared. If you remove one view instance, wouldn't this mean that when other view instances are removed, the cleanup fns would be gone?  Let's add a test for this ?

**Review by kara — CHANGES_REQUESTED** [review]

**Review by kara — APPROVED** [review]
LGTM

**Comment by kara** [comment]
[presubmit](http://test/OCL:229851752:BASE:229851780:1547776542391:8b4e3ccb)

**Comment by alxhub** [comment]
Considering @kara and @mhevery's approval to be global.
